### PR TITLE
combine sort filter all views

### PR DESCRIPTION
**Pull request by pgandhizaizi**
## Changes in this PR

1. created new components for sorting lists and date filters
2. combined sort and filter functionality in the browse, transferring body and series view
3. added test cases for sorting+filter and end-to-end test cases for browse, transferring body and series view
4. updated readme.md to add additional parameter when running playwright codegen to ignore https errors

## JIRA ticket

AYR 707

### After

- [ ] Requires env variable(s) to be updated

### CODE REVIEW

File: e2e_tests/test_browse.py [hunk 1]
```diff
@@ -10,25 +10,318 @@ def test_has_title(authenticated_page: Page):
...
-def test_sort_dropdown(authenticated_page: Page):
-    authenticated_page.goto("/browse")
-    authenticated_page.select_option("#sort", "body-a")
-    authenticated_page.select_option("#sort", "body-b")
-    authenticated_page.select_option("#sort", "series-a")
-    authenticated_page.select_option("#sort", "series-b")
-    authenticated_page.select_option("#sort", "date-first")
-    authenticated_page.select_option("#sort", "date-last")
+class TestBrowse:
+    def test_browse_sort_dropdown(self, authenticated_page: Page):
```
Comment (anthonyhashemi) [inline e2e_tests/test_browse.py]: tests are looking good.

I'd just suggest that we could maybe get away without needing these sort dropdown e2e tests. 
I cant see much benefit since the dropdown is tested as part of the sort and filter e2e tests and then all the edge cases of the options are tested in all the standard pytest tests you have written

File: e2e_tests/test_browse.py [hunk 2]
```diff
@@ -10,25 +10,318 @@ def test_has_title(authenticated_page: Page):
...
+            "Transferring body",
+            "Series",
+            "Last record transferred",
+            "Records held",
+            "Consignments within series",
+        ]
+
+        assert rows.count() == 1
+
+        if cols.count() > 0:
```
Comment (anthonyhashemi) [inline e2e_tests/test_browse.py]: we should not have conditional statements in tests especially around assertions as this will mean we cant trust 2 runs of the test behave the same.

There's no reason cols.count should be 0, if it is there is a problem and we want the test to error out and let us know there is a problem. Currently it will silently (and incorrectly) pass

File: app/tests/test_browse.py [hunk 3]
```diff
@@ -234,6 +234,152 @@ def test_browse_get_without_filter(
         verify_browse_view_header_row(response.data)
         verify_data_rows(response.data, expected_rows)
 
+    def test_browse_get_with_transferring_body_filter_and_transferring_body_sorting_z_to_a(
```
Comment (anthonyhashemi) [inline app/tests/test_browse.py:237]: As suggested before, I think we could reduce a lot of this test code by paramterizing the query parameters passed into the client.get request and the expected rows

**Review by anthonyhashemi — CHANGES_REQUESTED** [review]
Generally looks really good. Have left some suggestions about about reducing size of test code and removing potentially unnecessary e2e tests.

Have requested changes specifically due to the conditional statements in tests, as explained in the comment.

File: e2e_tests/test_browse.py (hunk 1 above)
Comment (pgandhizaizi) [reply e2e_tests/test_browse.py]: removed sort dropdown test cases as requested

File: e2e_tests/test_browse.py (hunk 2 above)
Comment (pgandhizaizi) [reply e2e_tests/test_browse.py]: removed condition as requested

File: app/tests/test_browse.py (hunk 3 above)
Comment (pgandhizaizi) [reply app/tests/test_browse.py:237]: we need to look during the refactor and work out best way to reduce the test cases

**Review by anthonyhashemi — APPROVED** [review]
Nice one, looks good 👌
